pkg/command/map_string.go

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of Cilium
+
+package command
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+
+	"github.com/spf13/cast"
+	"github.com/spf13/viper"
+)
+
+// GetStringMapString contains one enhancement to support k1=v2,k2=v2 compared to original
+// implementation of GetStringMapString function
+func GetStringMapString(vp *viper.Viper, key string) map[string]string {
+	return getStringMapString(vp, key, ",", "=")
+}
+
+func getStringMapString(vp *viper.Viper, key string, pairSeparator string, keyValueSeparator string) map[string]string {
+	data := vp.Get(key)
+	v, err := cast.ToStringMapStringE(data)
+	if err != nil {
+		var syntaxErr *json.SyntaxError
+		if errors.As(err, &syntaxErr) {
+			switch s := data.(type) {
+			case string:
```

Comment:
I think there is some potential of malformed json to not mix well with a value `=` here? Perhaps covering something like `{"foo": "bar",=asdf` would put me more at ease

Reply:
thanks, I have added a test case for the same.